ios/Classes/SwiftCloudKitPlugin.swift

```diff
@@ -45,93 +45,97 @@ public class SwiftCloudKitPlugin: NSObject, FlutterPlugin {
          } else {
             result(FlutterError.init(code: "Error", message: "Cannot pass key and value parameter", details: nil))
          }
-        
+
     case "get":
-        
+
         if let arguments = call.arguments as? Dictionary<String, Any>,
            let key = arguments["key"] as? String,
            let containerId = arguments["containerId"] as? String {
-            
+
             let database = CKContainer(identifier: containerId).privateCloudDatabase
 
-           
+
             let query = CKQuery(recordType: "StorageItem", predicate: NSPredicate(value: true))
-            
+
             database.perform(query, inZoneWith: nil) { (records, error) in
 
                 let foundRecords = records?.compactMap({ $0.value(forKey: key) as? String })
-                
+
                 result(foundRecords)
-        
+
             }
-            
+
          } else {
             result(FlutterError.init(code: "Error", message: "Cannot pass key and value parameter", details: nil))
          }
-        
+
     case "delete":
-        
+
         if let arguments = call.arguments as? Dictionary<String, Any>,
            let key = arguments["key"] as? String,
            let containerId = arguments["containerId"] as? String {
-            
-            let database = CKContainer(identifier: containerId).privateCloudDatabase
 
-           
+            let database = CKContainer(identifier: containerId).privateCloudDatabase
             let query = CKQuery(recordType: "StorageItem", predicate: NSPredicate(value: true))
-            
+            var deletedRecord = false
+
             database.perform(query, inZoneWith: nil) { (records, error) in
-                
+                let deleteGroup = DispatchGroup()
+
                 records?.forEach({ (record) in
-                    
                     if record.value(forKey: key) != nil {
+                        deleteGroup.enter()
                         database.delete(withRecordID: record.recordID) { (recordId, error) in
-                            
+                            if recordId != nil, error == nil {
+                                deletedRecord = true
+                            }
+                            deleteGroup.leave()
                         }
-                        
                     }
-                    
-                    
                 })
-                
-                
+
+                deleteGroup.notify(queue: .main) {
+                    if deletedRecord {
+                        result(true)
```

Comment:
Are you returning now an boolean if the deletion was successfully? Then you also need to adapt the return type of the delete function in the cloud_kit.dart file.